core/domain/services/capabilities/CapabilitiesChecker.php

```diff
@@ -48,20 +48,25 @@ protected function capabilities()
      * If any of the individual capability checks fails, then the command will NOT be executed.
      *
      * @param CapCheckInterface|CapCheckInterface[] $cap_check
+     * @param bool                                  $suppress_exceptions
      * @return bool
      * @throws InvalidClassException
      * @throws InsufficientPermissionsException
      */
-    public function processCapCheck($cap_check)
+    public function processCapCheck($cap_check, bool $suppress_exceptions = false): bool
     {
         if (is_array($cap_check)) {
+            $passed = true;
             foreach ($cap_check as $check) {
-                $this->processCapCheck($check);
+                $passed = $this->processCapCheck($check) ? $passed : false;
             }
-            return true;
+            return $passed;
```

Comment:
we initially set `$passed` to `true`, 
then if any cap check fails, it will get toggled to `false`